### PR TITLE
Bug 1001757 - Allow building core apps outside of profile on b2g desktop

### PR DESCRIPTION
**Pull request by qdot**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[qdot Kyle Machulis (qdot)](https://github.com/qdot) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=e587b61c0bf9)

**Comment by try-server-hook** [comment]
[qdot Kyle Machulis (qdot)](https://github.com/qdot) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=1b473bdeb223)

**Comment by try-server-hook** [comment]
[qdot Kyle Machulis (qdot)](https://github.com/qdot) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=bfc04b945ca2)

**Comment by try-server-hook** [comment]
[qdot Kyle Machulis (qdot)](https://github.com/qdot) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4eb2e7a1b301)

File: build/preferences.js [hunk 1]
```diff
@@ -177,6 +177,13 @@ PreferencesBuilder.prototype.preparePref = function() {
   if (this.config.RAPTOR === '1') {
     this.userPrefs['dom.performance.enable_user_timing_logging'] = true;
   }
+
+  // If this is already equal to the profile directory, no need to
+  // write the preference
+  if (this.config.COREWEBAPPS_DIR != this.config.PROFILE_DIR) {
```
Comment (RickyChien) [inline build/preferences.js]: ```!==``` is better

**Comment by try-server-hook** [comment]
[qdot Kyle Machulis (qdot)](https://github.com/qdot) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=683da6f41510)

**Pull request merged**